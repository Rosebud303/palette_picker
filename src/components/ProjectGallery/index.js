import React from 'react';
import uniqid from 'uniqid';
import Project from '../Project';


const Gallery = () => {
  // set up for redux
  // map through projects, return a project component
  const mockProjects = [
    {
      name: 'Fred',
      palettes: [{
        name: 'Fancy',
        color_1: '#F655AA',
        color_2: '#FBA904',
        color_3: '#7AFDDF',
        color_4: '#571E7F',
        color_5: '#0071BD'
      },
      {
        name: 'Dazzle',
        color_1: '#32213A',
        color_2: '#383B53',
        color_3: '#66717E',
        color_4: '#D4D6B9',
        color_5: '#D1CAA1'
      }

      ]
    },
    {
      name: 'Bob',
      palettes: [{
        name: 'Fancy',
        color_1: '#F655AA',
        color_2: '#FBA904',
        color_3: '#7AFDDF',
        color_4: '#571E7F',
        color_5: '#0071BD'
      },
      {
        name: 'Dazzle',
        color_1: '#32213A',
        color_2: '#383B53',
        color_3: '#66717E',
        color_4: '#D4D6B9',
        color_5: '#D1CAA1'
      }
      ]
    }
  ];
  const projects = (unicorns) => {
    const key = uniqid();
    const lilUnis = unicorns.map(uni => (
      <Project
        key={key}
        name={uni.name}
        palettes={uni.palettes}
      />
    ));
    return lilUnis;
  };

  return (
    <section className="project-gallery">
      { projects(mockProjects) }
    </section>
  );
};

export default Gallery;
