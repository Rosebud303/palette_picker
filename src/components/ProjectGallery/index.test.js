/* eslint-disable import/no-named-as-default */
import React from 'react';
import { shallow } from 'enzyme';
import ProjectGallery from '.';

describe('ProjectGallery', () => {
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<ProjectGallery />);

    expect(wrapper).toMatchSnapshot();
  });
});
