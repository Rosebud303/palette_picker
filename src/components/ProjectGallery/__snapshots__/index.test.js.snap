// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectGallery should match the snapshot 1`] = `ShallowWrapper {}`;
